### PR TITLE
added build doc

### PR DESCRIPTION
**Pull request by Yurlungur**
## PR Summary

The discussion in issue #111 indicates to me that there's a concensus that we'd like to document common builds, in addition to adding machine-specific settings in the build system (a la MR #114). I therefore went ahead and got the ball rolling on this. I added a build doc into `docs` for my virtual machine. 

As we get things working on various machines that we have trouble with, let's add instructions here. 

Tagging interested people so you're aware, and in case you object. I asked @pgrete and @jlippuner to review since they've been active participants on issue #111. But anyone with skin in the game, feel free to comment.

@gshipman @junghans  please let me know if adding build instructions raises security concerns.

tagging:
@jdolence @nmsriram @forrestglines @AndrewGaspar @jmstone @JoshuaSBrown 

## PR Checklist

- [x] Code passes cpplint (no new code)
- [x] New features are documented (no new features).
- [x] Adds a test for any bugs fixed. Adds tests for new features (no new features).
- [x] Code is formatted (no code)

### CODE REVIEW

File: docs/building.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# Clone parthenon, with submodules
+git clone --recursive https://github.com/lanl/parthenon.git
+# make a bin directory
+mkdir bin
+cd bin
+# configure and build
+cmake ..
+make -j
+# run unit tests
+ctest -LE performance
```
Comment (JoshuaSBrown) [inline docs/building.md:22]: This will also run the regression tests.

File: docs/building.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+install cmake build-essentials libhdf5-openmpi-dev
+
+# Clone parthenon, with submodules
+git clone --recursive https://github.com/lanl/parthenon.git
+# make a bin directory
+mkdir bin
+cd bin
+# configure and build
+cmake ..
+make -j
```
Comment (junghans) [inline docs/building.md:20]: `cmake -j  --build .`

File: docs/building.md (hunk 1 above)
Comment (Yurlungur) [reply docs/building.md:22]: Good catch.

File: docs/building.md (hunk 2 above)
Comment (Yurlungur) [reply docs/building.md:20]: Good catch.

File: docs/building.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+install cmake build-essentials libhdf5-openmpi-dev
+
+# Clone parthenon, with submodules
+git clone --recursive https://github.com/lanl/parthenon.git
+# make a bin directory
+mkdir bin
+cd bin
+# configure and build
+cmake ..
+make -j --build .
```
Comment (junghans) [inline docs/building.md:20]: `cmake` not `make`.

Comment (Yurlungur) [reply docs/building.md:20]: Oops. not sure how I managed that. I'll fix it.

Comment (junghans) [reply docs/building.md]: Huh?

I really are two cmake call:
```
cmake ..
cmake --build .
```

Comment (Yurlungur) [reply docs/building.md]: Oh I see. Sorry, I misunderstood.

**Comment by Yurlungur** [comment]
It's been 24 hours and people seem happy. Pulling the trigger.

**Pull request merged**